Document/0x05d-Testing-Data-Storage.md

```diff
@@ -509,21 +421,181 @@ There are several different open-source libraries that offer encryption capabili
+- Identify if SQLite databases are encrypted. If so, determine how the database password is generated and stored and if this is sufficiently protected as described in the "[Storing a Key](#storing-a-key)" section of the Keystore overview.
+- Check Shared Preferences that are stored as XML files (in `/data/data/<package-name>/shared_prefs`) for sensitive information. Avoid using Shared Preferences and other mechanisms that can't protect data when you are storing sensitive information. Shared Preferences is insecure and unencrypted by default. You can use [secure-preferences](https://github.com/scottyab/secure-preferences "Secure-preferences encrypts the values of Shared Preferences") to encrypt the values stored in Shared Preferences, but the Android KeyStore should be your first choice for storing data locally securely.
 - Check the permissions of the files in `/data/data/<package-name>`. Only the user and group created when you installed the app (e.g., u0_a82) should have user read, write, and execute permissions (`rwx`). Other users should not have permission to access files, but they may have execute permissions for directories.
+- Check for the usage of any Firebase Real-time databases and attempt to identify if they are misconfigured by making the following network call:
+  - `https://_firebaseProjectName_.firebaseio.com/.json`
 - Determine whether a Realm database is available in `/data/data/<package-name>/files/`, whether it is unencrypted, and whether it contains sensitive information. By default, the file extension is `realm` and the file name is `default`. Inspect the Realm database with the [Realm Browser](https://github.com/realm/realm-browser-osx "Realm Browser for macOS").
-- Check external storage for data. Don't use external storage for sensitive data because it is readable and writeable system-wide.
 
-Files saved to internal storage are by default private to your application; neither the user nor other applications can access them. When users uninstall your application, these files are removed.
+In general sensitive data stored locally on the device should always be at least encrypted, and any keys used for encryption methods should be securely stored within the Android Keystore. These files should also be stored within the application sandbox.
+
+Even more ideally, if achievable, sensitive data shouldn't be stored on the device at all and instead should be stored off device.
```

Comment:
do we need such a final conclusion, what if we give all recommendations already in the overview of the test case and here only instructuons on what to test? Similar as what you did in line 627.

Reply:
I'll move this into the overview; I think its nice to have a one liner general recommendation on things where possible to help summarize things and also operate as a reminder.